client/k8srestconfig/k8s_rest_config.go

```diff
@@ -107,8 +110,14 @@ func New(config Config) (*rest.Config, error) {
 	}
 
 	// Settings.
-	if config.Address == "" && !config.InCluster {
-		return nil, microerror.Maskf(invalidConfigError, "%T.Address must not be empty when not creating in-cluster client", config)
+	if config.Address == "" && !config.InCluster && config.KubeConfig == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Address must not be empty when not using %T.InCluster or %T.KubeConfig", config)
+	}
+	if config.Address != "" && config.KubeConfig != "" {
+		return nil, microerror.Maskf(invalidConfigError, "cannot use %T.Address and %T.KubeConfig", config)
+	}
+	if config.InCluster && config.KubeConfig != "" {
+		return nil, microerror.Maskf(invalidConfigError, "cannot use %T.InCluster and %T.KubeConfig", config)
```

Comment:
I updated the error handling. Adding a unit test didn't go well because the env vars for in cluster were missing.

We do have some coverage via the new Service test in the operators.

Reply:
We should anyway check how we can have `Validate() error` on `Config` structures. This is then stateless and can be unit tested. The constructor is then also cleaner. I was yet not decided on how to structure that though.

Reply:
Having Validate for this will be cool. It does help the testing situation a lot. 

For now I want to move this forward for now so we can start using Kind.